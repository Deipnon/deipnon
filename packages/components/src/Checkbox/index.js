// @flow

import * as React from 'react'
import { css } from '@emotion/core'
import styled from '@emotion/styled'

const checkboxInputStyle = css`
	box-sizing: border-box;
	padding: 0;
	position: absolute;
    margin-left: -1.25rem;
`
const checkboxFormStyle = css`
	position: relative;
	display: block;
	padding-left: 1.25rem;
`
const checkboxLabelStyle = css`
	margin-bottom: 0;
`

const CheckboxForm = styled('div')(checkboxFormStyle)
const CheckboxInputElement = styled('input')(checkboxInputStyle)
const CheckboxLabelElement = styled('label')(checkboxLabelStyle)

type PropsType = {|
	checked: ?boolean,
	children: React$Node,
	onChange: (e: SyntheticEvent<HTMLInputElement>) => void
|}

const Checkbox = (props: PropsType) => {
	return (
		<CheckboxForm>
			<CheckboxLabelElement>
				<CheckboxInputElement type="checkbox" checked={props.checked} onChange={props.onChange} />
				{props.children}
			</CheckboxLabelElement>
		</CheckboxForm>
	)
}

export default Checkbox
