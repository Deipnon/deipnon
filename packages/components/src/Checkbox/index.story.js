import React from 'react'
import { storiesOf } from '@storybook/react'
import { withKnobs } from '@storybook/addon-knobs/react'

import Checkbox from '.'

const stories = storiesOf('Checkbox', module)
stories.addDecorator(withKnobs)

class CheckboxSample extends React.PureComponent {
	constructor (props) {
		super(props)
		this.state = {
			isChecked: false
		}
	}

	handleChange = (e) => {
		this.setState({ isChecked: !this.state.isChecked })
	}

	render () {
		const { state: { isChecked }, handleChange } = this
		return (
			<Checkbox onChange={handleChange} checked={isChecked}>Label</Checkbox>
		)
	}
}

stories.add('default', () => <CheckboxSample />)
