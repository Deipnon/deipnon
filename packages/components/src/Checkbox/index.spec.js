import React from 'react'
import renderer from 'react-test-renderer'
import 'jest-emotion'

import Checkbox from '.'

describe('Checkbox', () => {
	test('renders correctly', () => {
		const tree = renderer.create(<Checkbox onChange={() => {}}/>).toJSON()
		expect(tree).toMatchSnapshot()
	})
})
