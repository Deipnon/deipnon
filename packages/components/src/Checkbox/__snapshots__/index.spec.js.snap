// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox renders correctly 1`] = `
<div
  className="css-17u2vr1"
>
  <label
    className="css-x823z2"
  >
    <input
      checked="false"
      className="css-5p61z4"
      onChange={[Function]}
      type="checkbox"
    />
    Remember Me
  </label>
</div>
`;
